pydantic/_internal/_generate_schema.py

```diff
@@ -768,23 +769,66 @@ def apply_validators(schema: core_schema.CoreSchema, validators: list[Validator]
     return schema
 
 
+_SerializerType = Literal[
+    'field-wrap',
+    'general-wrap',
+    'field-plain',
+    'general-plain',
+]
+
+
+def _is_unbound_instance_method(sig: Signature) -> bool:
+    if len(sig.parameters) < 1:
+        raise TypeError
```

Comment:
```suggestion
```
Looks like we don't need this from the usage below.

Reply:
Well we are doing `next(iter(thing))` here so I like having the check that we have at least 1 item here to justify that we can call `next(iter(...))` without catching the failure for an empty iterable